enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,350 @@
+      keydata: LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUlJQ0lqQU5CZ2txaGtpRzl3MEJBUUVGQUFPQ0FnOEFNSUlDQ2dLQ0FnRUEzQzJlVGdJQUo3aGxveDdDSCtIcE1qdDEvbW5lYXcyejlHdE9NUmlSaEgya09ZalRadGVLSEtnWUJHcGViajRBcUpWYnVRaWJYZTZKYVFHQUFER0VOZXozTldsVXpCby9FUUEwaXJDRnN6dlhVbTE2cWFZMG8zOUZpbWpsVVovaG1VNVljSHhxMzR2OTh4bGtRbUVxekowR0VJMzNtWTFMbWFEM3ZhYmd3WWcwb3lzSTk1Z1V1Tk81TmdZUHA4WDREaFNoSmtyVEl5dDJLTEhYWW5BMExzOEJlbG9PWVJlTnJhZmxKRHNzaE5VRFh4MDJhQVZSd2RjMXhJUDArRTlZaTY1ZE4zKzlReVhEOUZ6K3MrTDNjZzh3bDdZd3ZZb1Z2NDhndklmTHlJbjJUaHY2Uzk2R0V6bXBoazRjWDBIeitnUkdocWpyajU4U2hSZzlteitrcnVhR0VuVGcyS3BWR0gzd3I4Z09UdUFZMmtqMnY1YWhnZWt4V1pFN05vazNiNTBKNEpnYXlpSnVSL2R0cmFQMWVMMjlFMG52akdsMXptUXlGNlZnNGdIVXYwaktrcnJ2QUQ4c1dNY2NBS00zbXNXU01uRVpOTnljTTRITlNobGNReG5xU1lFSXR6MGZjajdYamtKbnAxME51Z2lVWlNLeVNXOHc0R3hTaFNraGRGbzByRDlkVElRZkJoeS91ZHRQWUkrK2VoK243QTV2UVV4Wk5BTmZqOUhRbC81Z3lFbFV6TTJOekJ2RHpHellSNVdVZEVEaDlJQ1I4ZlFpMVIxNUtZU0h2Tlc3RW5ucDdZT2d5dmtoSkdwRU5PQkF3c1pLMUhhMkJZYXZMMk05NDJzSkhxOUQ1eEsrZyszQU81eXp6V2NqaUFDMWU4RURPcUVpY01Ud05LOENBd0VBQVE9PQotLS0tLUVORCBQVUJMSUMgS0VZLS0tLS0K
+```
+
+#### Installer
+
+As discussed in [the *Proposal* section](#proposal), the installer will grow a new, feature-gated `internalTestingImagePolicy` property driving ClusterVersion `overrides` extension.
+When set to `Disable`, the following `overrides` entry will be added to the ClusterVersion manifest:
+
+```yaml
+kind: ClusterImagePolicy
+group: config.openshift.io
+name: openshift
+unmanaged: true
+```
+
+##### Installer callers
+
+Hive, the assisted installer, and other actors who create install-configs will need to learn to set `internalTestingImagePolicy: Disabled` if they wish to test unsigned releases that enable the feature-gate.
+
+#### Implementation order
+
+To reduce risk, the recommended implementation order is:
+
+1. Machine-config operator [lifts the release payload scope restriction](#api-extensions).
+2. [Parallel Cincinnati work][OTA-1267] allows ART to push Sigstore signatures to `quay.io/openshift-release-dev/ocp-release:sha256-....sig`.
+3. Testing [the planned ClusterImagePolicy](#machine-config-operator) confirms that the approach works, possibly including [restricted-network use](#restricted-network-use).
+    If testing turns up gaps, round to see if we want to pivot or abandon this effort.
+4. [Installer `internalTestingImagePolicy` property](#installer).
+5. Tooling interacting with unsigned `quay.io/openshift-release-dev/ocp-release` release images ([CI, QE, assisted-installer, Hive, others?](#installer-callers)) updated to set `internalTestingImagePolicy`.
+6. Machine-config operator [adds the `openshift` ClusterImagePolicy manifest](#machine-config-operator).
+
+### Risks and Mitigations
+
+#### Changing our minds
+
+Not much risk for connected clusters.
+If we decide that the policy is not helpful, we can remove from future releases via a [deletion annotation](/enhancements/update/object-removal-manifest-annotation.md).
+
+#### Signing key compromise
```

Comment:
I'm certainly no Sigstore expert, but I'd have expected a compromised key could be used to generate malicious signatures which could be inserted into either quay.io or a mirror (for clusters with an ImageDigestMirrorSet set up).  [The Sigstore threat model][1] includes:

> This means that secure distribution of Sigstore key material is paramount for security—if that process can be intercepted, we can consider all of the infrastructure effectively compromised.

but me quoting one line out of context is a pretty handwavy kind of security argument.  Thoughts on who could audit this enhancement from a security perspective to provide a more informed take than me guessing and skimming brief docs posts?

[1]: https://docs.sigstore.dev/about/threat-model/#secure-distribution-and-revocation-of-sigstore-key-material

Reply:
> but I'd have expected a compromised key could be used to generate malicious signatures which could be inserted into either quay.io or a mirror

It could. Just suggesting that an attacker would need additional access to push the signature to a registry (or elevated permissions on the target cluster) in order to achieve an attack.

> This means that secure distribution of Sigstore key material is paramount for security—if that process can be intercepted, we can consider all of the infrastructure effectively compromised.

I don't disagree and this is the assumption we would work under.